Belos: lots of warnings regarding usage of null pointers

## Bug Report

Apologies if this was already reported (though I did not find it among issues with the "pkg: belos" label).

I am getting several compiler warnings from a few Belos files (I think they are all `BelosXYZSolMgr.hpp`). The warnings are of the form
```
/ascldap/users/lbertag/workdir/trilinos/trilinos-src/develop/packages/belos/src/BelosBlockGmresSolMgr.hpp: In member function 'void Belos::BlockGmresSolMgr<ScalarType, MV, OP>::setParameters(const Teuchos::RCP<Teuchos::ParameterList>&) [with ScalarType = double; MV = Thyra::MultiVectorBase<double>; OP = Thyra::LinearOpBase<double>]':
/ascldap/users/lbertag/workdir/trilinos/trilinos-src/develop/packages/belos/src/BelosBlockGmresSolMgr.hpp:667:24: warning: 'this' pointer is null [-Wnonnull]
  667 |       paramsOrtho->set ("depTol", orthoKappa_ );
```
The compiler is indeed correct, since [here](https://github.com/trilinos/Trilinos/blob/6e2f44ec1e0ab10f0c483ad1274c6194349db76f/packages/belos/src/BelosBlockGmresSolMgr.hpp#L665) we are creating a null RCP, and dereferencing it inside the if statement, without first creating the new param list. The inline comment "can be null" seems to suggest the issue was somewhat known (even though it's an understatement: it _is_ null).

1. Sha:  6e2f44ec1e0ab10f0c483ad1274c6194349db76f
2. Compiler: g++ 13.2.0

I am not attaching the config/build scripts, since I think they are not really important: looking at the code, it seems obvious to me that the compiler is right. We should probably have something like this?
```c++
    Teuchos::RCP<Teuchos::ParameterList> paramsOrtho;   // can be null
    paramsOrtho = Teuchos::rcp(new Teuchos::ParameterList(""));
    if (orthoType_=="DGKS" && orthoKappa_ > 0) {
      paramsOrtho->set ("depTol", orthoKappa_ );
    }
```

I assume none of the Trilinos tests exercise the conditions that would trigger the if statement block to be executed?

--- a/packages/belos/src/BelosBlockCGSolMgr.hpp
+++ b/packages/belos/src/BelosBlockCGSolMgr.hpp
@@ -623,8 +623,9 @@ setParameters (const Teuchos::RCP<Teuchos::ParameterList> &params)
   // Create orthogonalization manager if we need to.
   if (ortho_ == Teuchos::null || changedOrthoType) {
     Belos::OrthoManagerFactory<ScalarType, MV, OP> factory;
-    Teuchos::RCP<Teuchos::ParameterList> paramsOrtho;   // can be null
+    Teuchos::RCP<Teuchos::ParameterList> paramsOrtho;   
     if (orthoType_=="DGKS" && orthoKappa_ > 0) {
+      paramsOrtho = Teuchos::rcp(new Teuchos::ParameterList());
       paramsOrtho->set ("depTol", orthoKappa_ );
     }
 

--- a/packages/belos/src/BelosBlockGmresSolMgr.hpp
+++ b/packages/belos/src/BelosBlockGmresSolMgr.hpp
@@ -662,8 +662,9 @@ void BlockGmresSolMgr<ScalarType,MV,OP>::setParameters( const Teuchos::RCP<Teuch
   // Create orthogonalization manager if we need to.
   if (ortho_ == Teuchos::null || changedOrthoType) {
     Belos::OrthoManagerFactory<ScalarType, MV, OP> factory;
-    Teuchos::RCP<Teuchos::ParameterList> paramsOrtho;   // can be null
+    Teuchos::RCP<Teuchos::ParameterList> paramsOrtho;   
     if (orthoType_=="DGKS" && orthoKappa_ > 0) {
+      paramsOrtho = Teuchos::rcp(new Teuchos::ParameterList());
       paramsOrtho->set ("depTol", orthoKappa_ );
     }
 

--- a/packages/belos/src/BelosPCPGSolMgr.hpp
+++ b/packages/belos/src/BelosPCPGSolMgr.hpp
@@ -616,6 +616,7 @@ void PCPGSolMgr<ScalarType,MV,OP,true>::setParameters( const Teuchos::RCP<Teucho
     Belos::OrthoManagerFactory<ScalarType, MV, OP> factory;
     Teuchos::RCP<Teuchos::ParameterList> paramsOrtho;   // can be null
     if (orthoType_=="DGKS" && orthoKappa_ > 0) {
+      paramsOrtho = Teuchos::rcp(new Teuchos::ParameterList());
       paramsOrtho->set ("depTol", orthoKappa_ );
     }
 

--- a/packages/belos/src/BelosPseudoBlockGmresSolMgr.hpp
+++ b/packages/belos/src/BelosPseudoBlockGmresSolMgr.hpp
@@ -786,6 +786,7 @@ setParameters (const Teuchos::RCP<Teuchos::ParameterList>& params)
     Belos::OrthoManagerFactory<ScalarType, MV, OP> factory;
     Teuchos::RCP<Teuchos::ParameterList> paramsOrtho;   // can be null
     if (orthoType_=="DGKS" && orthoKappa_ > 0) {
+      paramsOrtho = Teuchos::rcp(new Teuchos::ParameterList());
       paramsOrtho->set ("depTol", orthoKappa_ );
     }
 

--- a/packages/belos/tpetra/test/BlockGmres/CMakeLists.txt
+++ b/packages/belos/tpetra/test/BlockGmres/CMakeLists.txt
@@ -44,7 +44,10 @@ ENDIF()
 TRIBITS_ADD_EXECUTABLE_AND_TEST(
   Tpetra_BlockGMRES_hb_test
   SOURCES test_bl_gmres_hb.cpp
-  ARGS "--verbose"
+  ARGS 
+    "--verbose --ortho-type=DGKS"
+    "--verbose --ortho-type=ICGS"
+    "--verbose --ortho-type=IMGS"
   COMM serial mpi
   )
 

--- a/packages/belos/tpetra/test/BlockGmres/test_bl_gmres_hb.cpp
+++ b/packages/belos/tpetra/test/BlockGmres/test_bl_gmres_hb.cpp
@@ -156,6 +156,8 @@ int main(int argc, char *argv[]) {
     belosList.set( "Maximum Iterations", maxiters );       // Maximum number of iterations allowed
     belosList.set( "Convergence Tolerance", tol );         // Relative convergence tolerance requested
     belosList.set( "Orthogonalization", ortho );           // Orthogonalization type
+    if ( ortho == "DGKS" )
+      belosList.set( "Orthogonalization Constant",  1.41421356);
 
     int verbLevel = Belos::Errors + Belos::Warnings;
     if (debug) {

